cfpb/cfgov-refresh
# Set up split testing mechanism, and first test (Ask CFPB H1s)

Background: https://GHE/CFGOV/platform/issues/2812

These changes set up Wagtail pages to be able to be "split tested" – i.e., some pages are rendered differently. Our analytics team will track the pages we're experimenting on (grouped into "clusters") and see how the changes perform.

The first test is changing some Ask CFPB answer pages' main headings to use `<h1>` instead of `<h2>`.

## Additions

- A data file defining the clusters as groups of page IDs
  - This includes the first set of clusters, as provided to us by the analytics team.
- A new flag condition that takes the name of a cluster group and a `Page` object and checks to see if that `page.id` is found in that cluster group
- A new feature flag for the first experiment, `ASK_CFPB_H1`, used in the answer page template to set the main heading level to `h1` (rather than the current `h2`) on the answers in the clusters
  - The flag is on by default for answers in the clusters, and if successful, the flag can be given the boolean `True` condition to apply it to all answer pages until the flag is removed and the change is made permanent.
  - There is also an addition to the `AnswerPage` model class that overrides the default thing that the condition checks against the clusters – `page.id` – with the much more common way that we refer to Ask CFPB answers, their `answer.id`.

## Changes

- Removes a few `request` arguments from feature flag checks and the docs, because they are no longer required
- Something in me broke while working on this PR and I couldn't take the `base.py` linting errors in my editor anymore... so I fixed them. One small step towards un-ignoring our `settings` folder in flake8!
  - If you want to ignore the linting changes, the only substantive-to-this-PR change in `base.py` is the addition of the `ASK_CFPB_H1` flag (new lines 686-691).

## Testing

1. Pull branch
1. Visit http://localhost:8000/ask-cfpb/what-is-negative-amortization-en-103/ and see a larger main heading
   - Inspect it and see that it's an `h1`
1. Visit an answer that's not in the clusters, like http://localhost:8000/ask-cfpb/how-can-i-figure-out-if-i-can-afford-to-buy-a-home-and-take-out-a-mortgage-en-118/, and see that it's still rendering as an `h2`
1. Run the tests with `tox -e fast ask_cfpb.tests.models.test_pages` and `tox -e fast core.tests.test_feature_flags`

## Notes

- I don't remember if I was supposed to set the `h1` we're testing to display at the same size as `h2` or not. @kelleyholden, do you remember what we discussed?

## Checklist
